config/routes.php

```diff
@@ -101,7 +101,32 @@ function (RouteCollector $route): void {
 );
 
 // API
-$route->get('/api[/{resource:.+}]', 'ApiController@index');
+$route->addGroup(
+    '/api',
+    function (RouteCollector $route): void {
+        $route->get('', 'Api\IndexController@index');
+
+        $route->addGroup(
+            '/v0-beta',
+            function (RouteCollector $route): void {
+                $route->addRoute(['OPTIONS'], '[/{resource:.+}]', 'Api\IndexController@options');
+                $route->get('', 'Api\IndexController@indexV0');
+
+                $route->get('/angeltypes', 'Api\AngelTypeController@index');
+                $route->get('/news', 'Api\NewsController@index');
+                $route->get('/rooms', 'Api\RoomsController@index');
+                $route->get('/rooms/{room_id:\d+}/shifts', 'Api\ShiftsController@entriesByRoom');
+
+                $route->addRoute(
```

Comment:
I would say this should return a 404 for unknown resources.
And for the resources from above the response should be 405 for any method except `GET`.

501 should be used if the server implements some kind of protocol.

Reference in MDN: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501
- . The only methods that servers are required to support (and therefore that must not return 501) are [GET](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET) and [HEAD](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD).
- If the server does recognize the method, but intentionally does not support it, the appropriate response is [405 Method Not Allowed](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405).